spec/models/list_spec.rb

```diff
@@ -16,120 +16,76 @@
       .not_to change { list.webhook_token }
   end
 
-  describe "#next_user" do
+
+  describe "#ordered_users" do
     let(:list) { lists(:outofdate) }
 
-    it "is the first user when no previous users have been assigned" do
-      expect(list.tasks.size).to eq(0)
-      expect(list.next_user).to eq(users(:aldhsu))
+    it "returns alphabetical users when no tasks exist" do
+      expect(list.ordered_users).to eq([users(:aldhsu), users(:alex), users(:dave), users(:tate)])
     end
 
-    it "is the second user when the first user was previously assigned" do
+    it "returns reordered users when 1 task exists" do
       create_tasks(:aldhsu)
-      expect(list.tasks.size).to eq(1)
-      expect(list.next_user).to eq(users(:alex))
+      expect(list.ordered_users).to eq([users(:alex), users(:dave), users(:tate), users(:aldhsu)])
+    end
+
+    it "returns reordered users when many task exists" do
+      create_tasks(:aldhsu, :alex)
+      expect(list.ordered_users).to eq([users(:dave), users(:tate), users(:aldhsu), users(:alex)])
     end
 
-    it "is still the second user when the second user archived" do
+    it "returns alphabetical users when all users have tasks" do
+      create_tasks(:aldhsu, :alex, :dave, :tate)
+      expect(list.ordered_users).to eq([users(:aldhsu), users(:alex), users(:dave), users(:tate)])
+    end
+
+    it "returns the second user when they archived last" do
       _, second = create_tasks(:aldhsu, :alex)
       second.archived!
       expect(list.tasks.size).to eq(2)
-      expect(list.next_user).to eq(users(:alex))
+      expect(list.ordered_users).to eq([users(:alex), users(:dave), users(:tate), users(:aldhsu)])
     end
 
-    it "is also the second user when the first task has not been actioned" do
+    it "returns the second user when the first task has not been actioned" do
```

Comment:
Similar here:
```
it "is not affected by un-actioned tasks" do
```

Reply:
Actioned in 0b3bc8e18e5015f4f12370684ed2673d7dc519f4

Reply:
Find at:
0b3bc8e18e5015f4f12370684ed2673d7dc519f4